spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -360,25 +358,26 @@ private void deleteFiles(Iterable<String> locations) {
         .noRetry()
         .suppressFailureWhenFinished()
         .onFailure((location, exc) -> LOG.warn("Failed to delete: {}", location, exc))
-        .run(fileIO::deleteFile);
+        .run(location -> table.io().deleteFile(location));
   }
 
   private static ManifestFile writeManifest(
       List<Row> rows,
       int startIndex,
       int endIndex,
-      Broadcast<FileIO> io,
+      Broadcast<Table> tableBroadcast,
       String location,
       int format,
       Types.StructType combinedPartitionType,
       PartitionSpec spec,
       StructType sparkType)
       throws IOException {
 
+    FileIO io = tableBroadcast.value().io();
```

Comment:
since `io` is only used once, I think we can just put it inline at L380?

Reply:
Yes, sounds good, I made this change.